### ISSUE TITLE
Feature request: Swam carving transform

### ISSUE THREAD

**Issue by ternaus** [albumentations-team/albumentations#654]
Add Seam Carving transform for images, masks, bounding boxes, landmarks:

https://www.pyimagesearch.com/2017/01/23/seam-carving-with-opencv-python-and-scikit-image/

https://github.com/andrewdcampbell/seam-carving

**Comment by Bh4r4t** [comment]
@ternaus, I'd like to take up this enhancement task.

**Comment by KyloRen1** [comment]
Is it ok with the Albumentation policy to add algorithms covered by active patents? https://github.com/scikit-image/scikit-image/issues/3646

**Comment by ternaus** [comment]
@KyloRen1 This is a good question.

The legal story in the world of machine learning is not clear at all.

BatchNorm and DropOut are patented - we use them.

[Pytorch](https://github.com/pytorch/vision/issues/2597), [Tensorflow](https://github.com/tensorflow/models/issues/9131), [OpenVino](https://github.com/openvinotoolkit/open_model_zoo/issues/1433) teams have no clue if their pre-trained models could be used for commercial purposes.

I have mixed feelings and our opinions within the core team diverged.

Let's stay in the safe zone and not add the transform. We may revisit this if there will be evidence that it adds a lot of value to the model training.
